### ISSUE TITLE
Cannot unset string offsets in NewRelicHandler

### ISSUE THREAD

**Issue by jeremeamia** [Seldaek/monolog#1191]
I'm not sure if I've done something wrong or not, but I have what appears to be a normal logging configuration in a Laravel app with NewRelic as one of my handlers.

When it gets it [line 92 in the NewRelic handler](https://github.com/Seldaek/monolog/blob/master/src/Monolog/Handler/NewRelicHandler.php#L92), it goes 💥 with this message "PHP Fatal error:  Uncaught Error: Cannot unset string offsets".

I tried to debug this on one of the servers and I'm seeing that `$record['formatted']` is a string at this point, and not an array as the code expects. Is there a bug or have I messed up something somewhere? Any tips on tracking this down would be great.

Thanks.

**Comment by elazar** [comment]
`NewRelicHandler` inherits `handle()` from `AbstractProcessingHandler`, which calls `format()` on the formatter prior to calling the handler `write()` method. 

Basically, it looks like the only way for `$record['formatted']` to be a string is if `NormalizerFormatter->format()` is returning one. It proxies to `normalize()`, the documentation for which [says it can return a string](https://github.com/Seldaek/monolog/blob/334b8d8783a1262c3b8311d6599889d82e9cc58c/src/Monolog/Formatter/NormalizerFormatter.php#L89).

So, it looks like it may be a bug in `NewRelicHandler` where it's not handling that case properly?

**Comment by jeremeamia** [comment]
Yeah, I was trying to trace through that code too. That `normalize()` function is fun, because it's recursive. And though it can return a string, I'm not sure why it is in this case. It seems like `$record['formatted']` should still have `['context']`, but idk.

**Comment by elazar** [comment]
@jeremeamia Can you add a debugging line in `AbstractProcessingHandler->handle()` that checks the type of `$record['formatted']` and, if it's a string, runs [`debug_print_backtrace()`](http://php.net/manual/en/function.debug-print-backtrace.php) or something?

**Comment by jeremeamia** [comment]
Good idea. I'll dig in some more at some point today.

**Comment by jeremeamia** [comment]
I found my problem: https://github.com/illuminate/log/blob/master/LogManager.php#L354-L363

The Laravel `LogManager` sets `LineFormatter` as the formatter for every handler unless you explicitly configure `'formatter' => 'default'` in the configuration for the handler. Very unexpected.

So, I added `'formatter' => 'default'` to the handler config for my NewRelic handler, which then allowed it to use the `NormalizerFormatter` that it expects.
